### PR TITLE
Improvements for OS-kubespray resize

### PR DESCRIPTION
**Pull request by katyafervent**

### CODE REVIEW

File: kqueen/engines/openstack_kubespray.py [hunk 1]
```diff
@@ -334,10 +345,15 @@ def resize(self, node_count):
         node_count = int(node_count)
         master_count = len(self.cluster.metadata["resources"]["masters"])
         new_slave_count = node_count - master_count
-        if new_slave_count < 0:
-            return False, "Node count should be at least %s" % master_count
+        if new_slave_count <= 0:
+            return False, "Node count should be at least %s" % (master_count + 1)
```
Comment (naumvd95) [inline kqueen/engines/openstack_kubespray.py]: maybe ``Node amount should not be less that Master-node count ``

File: kqueen/engines/openstack_kubespray.py [hunk 2]
```diff
@@ -334,10 +345,15 @@ def resize(self, node_count):
...
         current_slave_count = len(self.cluster.metadata["resources"]["slaves"])
         delta = new_slave_count - current_slave_count
+
+        if delta == 0:
+            return False, "Cluster already has %s nodes" % node_count
+        self.cluster.state = config.CLUSTER_UPDATING_STATE
+        self.cluster.save()
```
Comment (naumvd95) [inline kqueen/engines/openstack_kubespray.py]: raw 361, maybe just else?

Comment (naumvd95) [reply kqueen/engines/openstack_kubespray.py]: elif redundant)

File: kqueen/engines/openstack_kubespray.py [hunk 3]
```diff
@@ -592,8 +607,11 @@ def _run_ansible(self, inventory="hosts.json", playbook="cluster.yml"):
                 env=env,
             )
             pipe.wait()
-            if pipe.returncode:
-                logger.warning("Non zero exit status from ansible (%s)" % pipe.returncode)
+        if pipe.returncode:
+            with open(self.ansible_log, "r") as log_file:
+                result = log_file.read()
+            if result.find("fatal:"):
```
Comment (naumvd95) [inline kqueen/engines/openstack_kubespray.py]: imo its bad case to find `fatal`. it looks like hardcode, can we trigger it on any other points? maybe exit code or smth similar?

File: kqueen/models.py [hunk 4]
```diff
@@ -38,26 +38,25 @@ class Cluster(Model, metaclass=ModelMeta):
     owner = RelationField(required=True, remote_class_name='User')
 
     def update_state(self):
+        # Check for stale clusters
+        max_age = timedelta(seconds=config.get('PROVISIONER_TIMEOUT'))
```
Comment (naumvd95) [inline kqueen/models.py]: what value do u prefer?

Comment (naumvd95) [reply kqueen/models.py]: imo it can depends on many reasons, need to define to properly

File: kqueen/engines/openstack_kubespray.py (hunk 3 above)
Comment (katyafervent) [reply kqueen/engines/openstack_kubespray.py]: no, it is ansible expected behavior. we check error code first and then fatal, there are no any names from user input, so it is ok to look for fatal with :

File: kqueen/models.py (hunk 4 above)
Comment (katyafervent) [reply kqueen/models.py]: it is defined in the config file now, I didn't change that part

File: kqueen/engines/openstack_kubespray.py (hunk 2 above)
Comment (katyafervent) [reply kqueen/engines/openstack_kubespray.py]: Didn't get you, I didnt change 361 raw. Are you suggesting to to rename elif to else?

File: kqueen/engines/openstack_kubespray.py (hunk 1 above)
Comment (katyafervent) [reply kqueen/engines/openstack_kubespray.py]: May be yes, we need a better message here

Comment (naumvd95) [reply kqueen/engines/openstack_kubespray.py]: agree, coz current message is unclear imo

File: kqueen/engines/openstack_kubespray.py (hunk 2 above)
Comment (naumvd95) [reply kqueen/engines/openstack_kubespray.py]: yep, i undrestand, its minor ) but if u have time )

File: kqueen/engines/openstack_kubespray.py (hunk 3 above)
Comment (naumvd95) [reply kqueen/engines/openstack_kubespray.py]: okok, agree

File: kqueen/models.py (hunk 4 above)
Comment (naumvd95) [reply kqueen/models.py]: ok, got it

**Pull request merged**